silkworm/db/snapshot_sync.cpp

```diff
@@ -135,6 +145,14 @@ bool SnapshotSync::download_snapshots(const std::vector<std::string>& snapshot_f
 
     std::latch download_done{num_snapshots};
     auto log_completed = [&](const std::filesystem::path& snapshot_file) {
+        // The same snapshot segment may be downloaded multiple times in case of content change over time and
+        // hence notified for completion multiple times. We need to count each snapshot segment just once here
+        static std::unordered_set<std::filesystem::path, PathHasher> snapshot_set;
```

Comment:
I'd move this before the lambda as a local variable in order to make SnapshotSync::download_snapshots callable multiple times